src/ln/channelmonitor.rs

```diff
@@ -1526,6 +1848,74 @@ impl ChannelMonitor {
 		}
 		false
 	}
+
+	pub(crate) fn is_resolving_output(&mut self, tx: &Transaction) -> Vec<(Option<HTLCSource>, Option<PaymentPreimage>, PaymentHash)> {
+		let mut htlc_updated = Vec::new();
+
+		let commitment_number = 0xffffffffffff - ((((tx.input[0].sequence as u64 & 0xffffff) << 3*8) | (tx.lock_time as u64 & 0xffffff)) ^ self.commitment_transaction_number_obscure_factor);
+		if commitment_number >= self.get_min_seen_secret() {
```

Comment:
We need to check that this is *actually* a remote commitment tx before we do this - its possible to have a false-positive and have a random 1-input transaction meet this check (or even have a local commitment tx that we broadcast meet this check!). Maybe could move this into check_spend_remote_commitment_tx?

Reply:
Yes, moved in check_spend_remote and add some comments on is_resolving_htlc_output